### PR TITLE
homework1811

### PR DESCRIPTION
**Pull request by vmaksimkina**

### CODE REVIEW

File: exercises/any_odds_and_even/main_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		{[]int{33, 11, 2}, true},
+		{[]int{4, 9, 3}, true},
+		{[]int{3, 7, 2}, true},
+		{[]int{-3, 1, 8}, true},
+		{[]int{-1, -3, -5}, false},
+		{[]int{-2, -3, 6}, true},
+		{[]int{-1, -5, 11}, false},
+	} {
+		got := AnyIsEven(tc.s)
+		if got != tc.want {
```
Comment (zayac) [inline exercises/any_odds_and_even/main_test.go:46]: nit: consider writing these lines as one:
```go
if got := AnyIsEven(ts.s); got != tc.want {
  ...
}
```

File: exercises/any_odds_and_even/main_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		}
+	}
+}
+
+func TestAnyIsEven(t *testing.T) {
+	for _, tc := range []struct {
+		s    []int
+		want bool
+	}{
+		{[]int{}, false},
```
Comment (zayac) [inline exercises/any_odds_and_even/main_test.go:30]: Please add tests with zeros.

File: exercises/any_odds_and_evens/main_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package main
+
+import "testing"
+
+func TestAnyIsOdd(t *testing.T) {
+	for _, tc := range []struct {
+		s    []int
+		want bool
+	}{
+		{[]int{}, false},
```
Comment (zayac) [inline exercises/any_odds_and_evens/main_test.go:10]: Please add tests with zeros and a test with with a slice containing a single negative integer.

File: exercises/filter_odds_and_evens/filter_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package main
+
+import "testing"
+
+func TestFilterOdds(t *testing.T) {
+	for _, tc := range []struct {
+		s    []int
+		want []int
+	}{
+		{[]int{-101}, []int{-101}},
```
Comment (zayac) [inline exercises/filter_odds_and_evens/filter_test.go:10]: None of the tests contain zeros. Could you please fix it?

File: exercises/rect/rect_test.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+			{1, 1, 1, 1, 1, 1, 1},
+		}},
+		{5, 5, [][]int{
+			{1, 1, 1, 1, 1},
+			{1, 0, 0, 0, 1},
+			{1, 0, 0, 0, 1},
+			{1, 0, 0, 0, 1},
+			{1, 1, 1, 1, 1},
+		}},
+		{9, 3, [][]int{
```
Comment (zayac) [inline exercises/rect/rect_test.go:31]: The dimensions are swapped. There should be 9 rows and 3 columns instead (you have 9 columns and 3 rows).

File: exercises/spiral/spiral_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+func TestSpiral(t *testing.T) {
+	for _, tc := range []struct {
+		x, y int
+		want [][]int
+	}{
+		{2, 2, [][]int{
+			{1, 2},
+			{4, 3},
+		}},
+		{4, 9, [][]int{
```
Comment (zayac) [inline exercises/spiral/spiral_test.go:17]: Same here. The dimensions are swapped.

File: exercises/spiral/spiral_test.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	"reflect"
+	"testing"
+)
+
+func TestSpiral(t *testing.T) {
+	for _, tc := range []struct {
+		x, y int
+		want [][]int
+	}{
+		{2, 2, [][]int{
```
Comment (zayac) [inline exercises/spiral/spiral_test.go:13]: Please add tests for dimensions equal to `1`.

File: exercises/rect/rect_test.go (hunk 5 above)
Comment (zayac) [reply exercises/rect/rect_test.go:31]: Actually, no. Please disregard the comment above.
